ProcessLib/HeatTransportBHE/BHE/BHEAbstract.h

```diff
@@ -0,0 +1,639 @@
+    };
+
+    struct Extern_Ra_Rb
+    {
+        /**
+         * whether or not using external given borehole thermal resistance
+         * values Ra, Rb
+         */
+        bool use_extern_Ra_Rb;
+
+        /**
+         * external given borehole internal thermal resistance value
+         */
+        double ext_Ra;
+
+        /**
+         * external given borehole thermal resistance value
+         */
+        double ext_Rb;
+    };
+
+    struct Extern_def_Thermal_Resistances
+    {
+        /**
+         * whether or not using user defined borehole thermal resistance Rfig,
+         * Rfog, Rgg, Rgs
+         */
+        bool if_use_defined_therm_resis;
+
+        /**
+         * external given borehole internal thermal resistance value
+         */
+        double ext_Rfig;
+
+        /**
+         * external given borehole internal thermal resistance value
+         */
+        double ext_Rfog;
+
+        /**
+         * external given borehole internal thermal resistance value
+         */
+        double ext_Rgg1;
+
+        /**
+         * external given borehole internal thermal resistance value
+         */
+        double ext_Rgg2;
+
+        /**
+         * external given borehole internal thermal resistance value
+         */
+        double ext_Rgs;
+    };
+
+    /**
+     * constructor
+     */
+    BHEAbstract(
+        const std::string name,
+        Borehole_Geometry borehole_geometry_,
+        Pipe_Parameters pipe_param_,
+        Refrigerant_Parameters refrigerant_param_,
+        Grout_Parameters grout_param_,
+        Extern_Ra_Rb extern_Ra_Rb_,
+        Extern_def_Thermal_Resistances extern_def_thermal_resistances_,
+        std::map<std::string,
+                 std::unique_ptr<MathLib::PiecewiseLinearInterpolation>> const&
+            bhe_curves,
+        BHE_BOUNDARY_TYPE my_bound_type =
+            BHE_BOUNDARY_TYPE::FIXED_INFLOW_TEMP_BOUNDARY,
+        bool if_use_ext_Ra_Rb = false,
+        bool user_defined_R_vals = false,
+        bool if_flowrate_curve = false,
+        int n_T_in = 1,
+        int n_T_out = 1)
+        : BHE_Net_ELE_Abstract(name, BHE::BHE_NET_ELE::BHE_NET_BOREHOLE, n_T_in,
+                               n_T_out),
+          _name(name),
+          bound_type(my_bound_type),
+          borehole_geometry(borehole_geometry_),
+          pipe_param(pipe_param_),
+          refrigerant_param(refrigerant_param_),
+          grout_param(grout_param_),
+          extern_Ra_Rb(extern_Ra_Rb_),
+          extern_def_thermal_resistances(extern_def_thermal_resistances_),
+          _bhe_curves(bhe_curves),
+          use_flowrate_curve(if_flowrate_curve),
+          if_use_ext_Ra_Rb(if_use_ext_Ra_Rb),
+          user_defined_R_vals(user_defined_R_vals){};
+
+    /**
+     * destructor
+     */
+    virtual ~BHEAbstract(){};
+
+    /**
+     * return the number of unknowns needed for this BHE
+     * abstract function, need to be realized.
+     */
+    virtual std::size_t get_n_unknowns() = 0;
```

Comment:
function names are also camel-case but starting with lower case letter. No underscores.
The names are preferably not shortened: `getNumberOfUnknowns()`.